lib/improver/utilities/temporal_interpolation.py

```diff
@@ -0,0 +1,419 @@
+        appropriately for use by iris' interpolation method.
+
+        Args:
+            initial_time (datetime.datetime):
+                The start of the period over which a time list is to be
+                constructed.
+            final_time (datetime.datetime).
+                The end of the period over which a time list is to be
+                constructed.
+
+        Returns:
+            list:
+                A list containing a tuple that specifies the coordinate and a
+                list of points along that coordinate to which to interpolate,
+                as required by the iris interpolation method:
+                    e.g. [('time', [<datetime object 0>,
+                                    <datetime object 1>])]
+
+        Raises:
+            ValueError: If list of times provided falls outside the range
+                        specified by the initial and final times.
+            ValueError: If the interval_in_minutes does not divide the time
+                        range up equally.
+        """
+        time_list = []
+        if self.times is not None:
+            self.times = sorted(self.times)
+            if self.times[0] < initial_time or self.times[-1] > final_time:
+                raise ValueError(
+                    'List of times falls outside the range given by '
+                    'initial_time and final_time. ')
+            time_list = self.times
+        else:
+            if ((final_time - initial_time).seconds %
+                    (60 * self.interval_in_minutes) != 0):
+                raise ValueError(
+                    'interval_in_minutes provided to time_interpolate does not'
+                    ' divide into the interval equally.')
+
+            time_entry = initial_time
+            while True:
+                time_entry = (time_entry +
+                              timedelta(minutes=self.interval_in_minutes))
+                if time_entry >= final_time:
+                    break
+                time_list.append(time_entry)
+
+        return [('time', time_list)]
+
+    @staticmethod
+    def check_cube_coords_dtype(cube, new_cube, dim_coords=None):
+        """
+        Update the data type of the coords within the new_cube
+        to those in the original cube. This can be limited to a list of
+        dimension coordinates and associated auxillary coordinates if required.
+
+        Args:
+        cube (iris.cube.Cube):
+            The original cube that will be used to checked against for the
+            required data type of that coordinate in the new_cube.
+        new_cube (iris.cube.Cube):
+            The cube that must be checked and adjusted using the coordinates
+            from the original cube.
+        dim_coords (list):
+            List of dimension coordinate names. This list will be used to
+            limit the correction of the coordinate data type
+            to these coordinates and associated auxillary coordinates.
```

Comment:
It looks like the data type correction will be limited to this list of coordinates only (the code doesn't search for associated aux coords).  Could you correct the docstring please.

Reply:
There is a check for associated auxillary coordinates see line 190. 
associated_coords = [crd.name() for crd in cube.coords()
                                         if cube.coord_dims(crd) == req_dim
                                         and crd.name() is not coord_name]
In the unit tests it updates time and forecast_period but not auxillary_coord.

Reply:
Ah OK.  Sorry, I got the loop mixed up.  This is fine.